File: 0728-self-dividing-numbers/0728-self-dividing-numbers.cpp
class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> a;
        for (int i=left; i<=right; i++){
            a.emplace_back(i);
        }
        vector<int> b;
        for (int i=0; i<a.size(); i++){
            int originum= a[i];
            int tempnum= a[i];
            bool checking= true;
            while (tempnum>0){
                int digit= tempnum%10;
                tempnum= tempnum/10;
                if ((digit==0) or (originum%digit!=0)){
                    checking= false;
                    break;
                }
            }
            if (checking){
                b.emplace_back(originum);
            }
        }
        return b;
    }
};

File: 0728-self-dividing-numbers/0728-self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        a=[]
        for i in range(left, right+1):
            a.append(i)
        d=[]
        for i in range(len(a)):
            originum= a[i]
            tempnum= a[i]
            checking= True
            while tempnum>0:
                digit=tempnum%10
                tempnum=tempnum//10
                if digit==0 or originum%digit!=0:
                    checking= False
                    break
            if checking:
                d.append(originum)
        return d